librad/src/git/include.rs

```diff
@@ -140,88 +161,114 @@ fn fetch_entry(remote: &Remote<LocalUrl>) -> (String, String) {
     )
 }
 
-impl<Path> TryFrom<Include<Path>> for Config
-where
-    Path: AsRef<path::Path>,
-{
-    type Error = git2::Error;
-
-    fn try_from(include: Include<Path>) -> Result<Self, Self::Error> {
-        let mut config = Self::new()?;
-        let file = include.file_path();
-        config.add_file(&file, ConfigLevel::Local, false)?;
-
-        for remote in include.remotes {
-            let (key, url) = url_entry(&remote);
-            config.set_str(&key, &url.to_string())?;
-
-            let (key, fetch) = fetch_entry(&remote);
-            config.set_str(&key, &fetch)?;
-        }
-
-        Ok(config)
-    }
-}
-
 #[cfg(test)]
 mod test {
-    use std::{io, path};
-
-    use git2::Config;
-
-    use super::*;
     use crate::{git::local::url::LocalUrl, hash::Hash, keys::SecretKey, peer::PeerId};
 
-    use librad_test::tempdir::WithTmpDir;
-
-    type TempInclude = WithTmpDir<Config>;
-
-    #[derive(Debug, thiserror::Error)]
-    enum Error {
-        #[error(transparent)]
-        Io(#[from] io::Error),
-        #[error(transparent)]
-        Git(#[from] git2::Error),
-    }
-
-    fn include<F>(constructor: F) -> Result<TempInclude, Error>
-    where
-        F: FnOnce(&path::Path) -> Include<&path::Path>,
-    {
-        WithTmpDir::new::<_, Error>(|path| {
-            let include = constructor(path);
-            let config = Config::try_from(include)?;
-            Ok(config)
-        })
-    }
+    use super::*;
 
     #[test]
-    fn can_create_trivial_include() -> Result<(), Error> {
+    fn can_create_and_update() -> Result<(), Error> {
```

Comment:
A few suggestions here:

* I'd like to see us moving more towards `lazy_static`'d keys where it does not actually matter
* Can the checks for presence of entries not be solved more concisely by iterating using [`entries`](https://docs.rs/git2/0.13.11/git2/struct.Config.html#method.entries) with a glob?
* What happens if you move the new file while still holding on to a live `Config`? Which version does it see?

Reply:
> * I'd like to see us moving more towards `lazy_static`'d keys where it does not actually matter

Which keys do you think would be better placed in `lazy_static`?

> * Can the checks for presence of entries not be solved more concisely by iterating using [`entries`](https://docs.rs/git2/0.13.11/git2/struct.Config.html#method.entries) with a glob?

That should be possible.

> * What happens if you move the new file while still holding on to a live `Config`? Which version does it see?

What is the scenario? Are there foreseeable cases where something holds onto the include file as `git2::Config`?